src/fw-update/fw-update-factory.cpp

```diff
@@ -50,10 +54,17 @@ namespace librealsense
                 auto usb = platform::usb_enumerator::create_usb_device(info);
                 if (!usb)
                     continue;
-                if (ds::RS_RECOVERY_PID == info.pid)
-                    return std::make_shared< ds_update_device >( shared_from_this(), usb );
-                if (ds::RS_USB2_RECOVERY_PID == info.pid)
-                    return std::make_shared< ds_update_device >( shared_from_this(), usb );
+                switch( info.pid )
+                {
+                case ds::RS_D400_RECOVERY_PID:
+                case ds::RS_D400_USB2_RECOVERY_PID:
+                    return std::make_shared< ds_d400_update_device >( shared_from_this(), usb, "D400");
```

Comment:
Why pass in `D400` if it's for `d400`?

Reply:
Right, changed to be hard coded inside as it use a base class functions and need to distinguish.